gstd/src/msg/macros.rs

```diff
@@ -64,6 +64,39 @@ macro_rules! impl_futures {
 
                 Ok(self)
             }
+
+            /// Execute a function when the reply is received
```

Comment:
add note that it runs in handle_reply and consumes the reply gas

Reply:
as a security note

Reply:
you also need to forbid double hooks applied to the same future

Reply:
added